File: frontend/src/services/post.service.ts
const API_BASE = import.meta.env.VITE_API_URL || 'http://localhost:3000';

async function makeRequest(endpoint: string, options: RequestInit = {}) {
  const token = localStorage.getItem('passport_buddy_token');
  const url = `${API_BASE}${endpoint}`;
  
  const response = await fetch(url, {
    credentials: 'include',
    headers: {
      'Content-Type': 'application/json',
      ...(token && { Authorization: `Bearer ${token}` }),
      ...options.headers,
    },
    ...options,
  });

  if (!response.ok) {
    const error = await response.json().catch(() => ({ message: 'Request failed' }));
    throw new Error(error.message || `HTTP ${response.status}`);
  }

  return response.json();
}

export const postService = {
  async toggleLike(postId: string): Promise<{ status: string; message: string }> {
    return makeRequest(`/api/v1/posts/${postId}/like`, { method: 'POST' });
  },

  async addComment(postId: string, content: string): Promise<{ status: string; message: string; comment: any }> {
    return makeRequest(`/api/v1/posts/${postId}/comment`, {
      method: 'POST',
      body: JSON.stringify({ content }),
    });
  },

  async deleteComment(postId: string, commentId: string): Promise<{ status: string; message: string }> {
    return makeRequest(`/api/v1/posts/${postId}/comment/${commentId}`, { method: 'DELETE' });
  },

  async deletePost(postId: string): Promise<{ status: string; message: string }> {
    return makeRequest(`/api/v1/posts/${postId}`, { method: 'DELETE' });
  },

  async incrementVideoViews(postId: string): Promise<{ status: string; message: string }> {
    return makeRequest(`/api/v1/posts/${postId}/views`, { method: 'POST' });
  },

  async createPost(data: FormData): Promise<{ status: string; message: string; post: any }> {
    const token = localStorage.getItem('passport_buddy_token');
    const response = await fetch(`${API_BASE}/api/v1/posts`, {
      method: 'POST',
      credentials: 'include',
      headers: {
        ...(token && { Authorization: `Bearer ${token}` }),
      },
      body: data,
    });

    if (!response.ok) {
      const error = await response.json().catch(() => ({ message: 'Request failed' }));
      throw new Error(error.message || `HTTP ${response.status}`);
    }

    return response.json();
  },

  async sharePost(postId: string): Promise<{ status: string; message: string }> {
    return makeRequest(`/api/v1/posts/${postId}/share`, { method: 'POST' });
  }
};

File: mobile/lib/services/post_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'package:shared_preferences/shared_preferences.dart';
import '../core/api/api_config.dart';

class PostService {
  static Future<String> get baseUrl async {
    final endpoint = await ApiConfig.discoverEndpoint();
    return endpoint.replaceAll('/graphql', '');
  }

  static Future<String?> _getAuthToken() async {
    final prefs = await SharedPreferences.getInstance();
    return prefs.getString('passport_buddy_token');
  }

  static Future<Map<String, String>> _getHeaders() async {
    final token = await _getAuthToken();
    return {
      'Content-Type': 'application/json',
      if (token != null) 'Authorization': 'Bearer $token',
    };
  }

  static Future<Map<String, dynamic>> toggleLike(String postId) async {
    try {
      final headers = await _getHeaders();
      final apiUrl = await baseUrl;
      final url = '$apiUrl/api/posts/$postId/like';
      
      final response = await http.post(
        Uri.parse(url),
        headers: headers,
      );

      if (response.statusCode == 200) {
        return json.decode(response.body);
      } else {
        throw Exception('Failed to toggle like: ${response.statusCode} - ${response.body}');
      }
    } catch (e) {
      throw Exception('Error toggling like: $e');
    }
  }
}